### PR TITLE
[4.0] TemplateAdapter prepared statements

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue # .

### Summary of Changes

Adds missing prepared statements and some cleanups to queries in `Joomla\CMS\Installer\Adapter\TemplateAdapter`.

### Testing Instructions

Test that installing, uninstalling and using different templates still works.

### Expected result

Works like before.

### Documentation Changes Required

No.

### CODE REVIEW

File: libraries/src/Installer/Adapter/TemplateAdapter.php [hunk 1]
```diff
@@ -181,21 +181,28 @@ protected function finaliseInstall()
...
+			->where(
+				[
+					$db->quoteName('s.template') . ' = :element',
+					$db->quoteName('s.client_id') . ' = :clientId',
+				]
+			);
 
-		$query = $db->getQuery(true)
-			->update($db->quoteName('#__menu'))
+		$query->bind(':element', $element)
```
Comment (wilsonge) [inline libraries/src/Installer/Adapter/TemplateAdapter.php:202]: We're missing the `strtolower` on this we were doing previously. Not sure it's going to make a massive difference or not however :/

Comment (SharkyKZ) [reply libraries/src/Installer/Adapter/TemplateAdapter.php:202]: It's always lowercase, as far as I can tell. Besides we had `strtolower()` for the same table column in one place but not in another so 🤷‍♂ . Tested myself on PostgreSQL and this part worked fine. But let's wait for human tests and hope someone else tests on PostgreSQL too.

Comment (alikon) [reply libraries/src/Installer/Adapter/TemplateAdapter.php:202]: test from a martian is the same ? :smile:

Comment (SharkyKZ) [reply libraries/src/Installer/Adapter/TemplateAdapter.php:202]: As long as they use a supported DB.

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on cbddc097cb6f5722faab9c8ef40fd31f118e4532<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26271">issues.joomla.org/tracker/joomla-cms/26271</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on cbddc097cb6f5722faab9c8ef40fd31f118e4532<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26271">issues.joomla.org/tracker/joomla-cms/26271</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26271">issues.joomla.org/tracker/joomla-cms/26271</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**